src/graphql/data/nft/Collection.ts

```diff
@@ -0,0 +1,127 @@
+import graphql from 'babel-plugin-relay/macro'
+import { Trait } from 'nft/hooks/useCollectionFilters'
+import { GenieCollection } from 'nft/types'
+import { useLazyLoadQuery } from 'react-relay'
+
+import { CollectionQuery } from './__generated__/CollectionQuery.graphql'
+
+const collectionQuery = graphql`
+  query CollectionQuery($address: String!) {
+    nftCollections(filter: { addresses: [$address] }) {
+      edges {
+        cursor
+        node {
+          bannerImage {
+            url
+          }
+          collectionId
+          description
+          discordUrl
+          homepageUrl
+          image {
+            url
+          }
+          instagramName
+          isVerified
+          name
+          numAssets
+          twitterName
+          nftContracts {
+            address
+            chain
+            name
+            standard
+            symbol
+            totalSupply
+          }
+          traits {
+            name
+            values
+            stats {
+              name
+              value
+              assets
+              listings
+            }
+          }
+          markets(currencies: ETH) {
+            floorPrice {
+              currency
+              value
+            }
+            owners
+            totalVolume {
+              value
+              currency
+            }
+            listings {
+              value
+            }
+            volume(duration: DAY) {
+              value
+              currency
+            }
+            volumePercentChange(duration: DAY) {
+              value
+              currency
+            }
+            floorPricePercentChange(duration: DAY) {
+              value
+              currency
+            }
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        hasPreviousPage
+        startCursor
+      }
+    }
+  }
+`
+export function useCollectionQuery(address: string): GenieCollection | undefined {
+  const queryData = useLazyLoadQuery<CollectionQuery>(collectionQuery, { address })
+
+  const queryCollection = queryData.nftCollections?.edges[0].node as any
```

Comment:
drop any cast for descriptive type

Reply:
Swapping this for wrapping all child elements in null checks